src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -1097,6 +1108,31 @@ void GcsActorManager::OnNodeDead(const NodeID &node_id,
   }
 }
 
+void GcsActorManager::SetPreemptedAndPublish(const NodeID &node_id) {
+  // The node is dead, so we need to mark all of its actors preempted.
```

Comment:
this comment seems wrong. 

```suggestion
  // Actors in the node is already dead. The actor death cause should've been already published, so we do nothing
```

Reply:
Yeah, it's wrong -- we set preempted before death and publish that state (when drain request arrives), then when it is actually dead we publish death.